app/components/thumbnail.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+const MAX_THUMBNAIL_DIMENSION = 999;
+
+export default class Thumbnail extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      failed: false,
+    };
+  }
+
+  getThumbnailSrc({ origin, width, height, src }) {
+    const srcPath = src.split('//').pop();
+    return (`${origin}/${width}x${height}/${srcPath}`);
+  }
+
+  handleError() {
+    if (!this.state.failed) {
```

Comment:
`this` is unbound here.  Adding `this.handleError = this.handleError.bind(this);` to the constructor fixes it